src/app.ts

```diff
@@ -166,7 +166,40 @@ export class App extends Construct {
       default:
         break;
     }
+
+  }
+
+  // return string with all yaml objects inside app
+  public synthYaml(): any {
+    // Since we plan on removing the distributed synth mechanism, we no longer call `Node.synthesize`, but rather simply implement
+    // the necessary operations. We do however want to preserve the distributed validation.
+    validate(this);
```

Comment:
Is there a way to reuse the code between `synth()` and `synthYaml()`? Seems like there's some duplication that will make it hard to maintain. Perhaps extract the common functionality into a private method?

Reply:
sure, any preference for the name?

Reply:
Tried something, let me know where you'd like changes.